dependency_cache.go

```diff
@@ -153,22 +150,46 @@ func customizeHttpClientTimeouts() (*HttpClientTimeouts, error) {
 	}, nil
 }
 
-// Returns the URI of a dependency mirror (optional).
-// Such mirror location can be defined in a binding of type 'dependency-mirror' with filename 'uri'
-// or using the environment variable 'BP_DEPENDENCY_MIRROR'. The latter takes precedence in case both are found.
-func getDependencyMirror(bindings libcnb.Bindings) (string, error) {
-	dependencyMirror := sherpa.GetEnvWithDefault("BP_DEPENDENCY_MIRROR", "")
-	// If no mirror was found in environment variables, try to find one in bindings.
-	if dependencyMirror == "" {
-		dependencyMirrorBindings, err := filterBindingsByType(bindings, "dependency-mirror")
-		if err == nil {
-			// Use the content of the file named "uri" as the mirror's URI.
-			dependencyMirror = dependencyMirrorBindings["uri"]
-		} else {
-			return "", err
+// Returns a key/value map with the URIs of all dependency mirrors. An empty map is returned if no mirrors are set.
+// Mirror locations can be defined in bindings of type 'dependency-mirror' or using env variables prefixed with 'BP_DEPENDENCY_MIRROR'.
+// Settings provided by env variables override those defined in bindings.
+func getDependencyMirrors(bindings libcnb.Bindings) (map[string]string, error) {
+	dependencyMirrors, err := filterBindingsByType(bindings, "dependency-mirror")
+	if err != nil {
+		return nil, err
+	}
+	dependencyMirrorsFromEnv := getDependencyMirrorsFromEnv()
+	for host, uri := range dependencyMirrorsFromEnv {
+		dependencyMirrors[host] = uri
+	}
+	return dependencyMirrors, nil
+}
+
+// Returns a key/value map of all dependency mirrors set in environment variables.
+func getDependencyMirrorsFromEnv() map[string]string {
+	mirrors := map[string]string{}
+	envs := os.Environ()
+	for _, env := range envs {
+		envPair := strings.SplitN(env, "=", 2)
+		hostnameSuffix, isMirror := strings.CutPrefix(envPair[0], "BP_DEPENDENCY_MIRROR")
+		hostnameEncoded, _ := strings.CutPrefix(hostnameSuffix, "_")
+		if isMirror {
+			mirrors[decodeHostnameEnv(hostnameEncoded)] = envPair[1]
 		}
 	}
-	return dependencyMirror, nil
+	return mirrors
+}
+
+// Takes an encoded hostname (from env key) and returns the decoded version in lower case.
+// Replaces double underscores (__) with one dash (-) and single underscores (_) with one period (.).
+func decodeHostnameEnv(encodedHostname string) string {
+	var decodedHostname string
+	if encodedHostname == "" {
```

Comment:
I think we should check and forbid someone from entering "default" as a hostname. If you did `BP_DEPENDENCY_MIRROR_DEFAULT`, I think, that would result in `encodedHostname` being `default`. I'm not sure that would be a problem exactly, but it would be defensive of us to skip that and just log a warning like "You have a mirror host name of `default`, which is prohibited. It has been ignored. Use `BP_DEPENDENCY_MIRROR` to set the default mirror.

Reply:
Has been added.